app-localize-behavior.html

```diff
@@ -117,6 +117,62 @@
     },
 
     /**
+     * Helper function to choose the most preferred language from a set of
+     * available languages. Supply a lookUp function, that takes a language
+     * and only returns it (or a variant) if you have it available, as well as
+     * an optional list of preferred languages, ordered from most-to-least
+     * preferred. If no preferred languages are given, the navigator object is
+     * checked for the browser's language preferences.
+     *
+     * For example, if your app is available in 'en-CA' and 'fr-CA', and the
+     * user's browser is set to prefer ['es-ES', 'fr-FR', 'en-GB'], you could
+     * do the following in order to calculate that 'fr-CA' is the best language
+     * that you have available for this user:
+     *
+     * const AVAILABLE_LANGUAGES = ['en-CA', 'fr-CA'];
+     *
+     * function lookUp(language) {
+     *   language = language.toLowerCase();
+     *   for (const availableLanguage of AVAILABLE_LANGUAGES) {
+     *     const parts = availableLanguage.toLowerCase().split('-');
+     *     while (parts.length) {
+     *       const joined = parts.join('-');
+     *       if (language === joined) {
+     *         return availableLanguage;
+     *       }
+     *       parts.pop();
+     *     }
+     *   }
+     * }
```

Comment:
(Since this `lookUp` function is pretty boilerplate too, I contemplated also including a `makeLookUpFunction(availableLanguages)` helper too, but I figured better to discuss this much first?)